python/hail/api1/dataset.py

```diff
@@ -2363,7 +2364,27 @@ def grm(self):
         :rtype: :py:class:`KinshipMatrix`
         """
 
-        return KinshipMatrix(Env.hail().methods.GRM.apply(self._jvds))
+        print("Computing GRM...")
+
+        vds = (self.annotate_variants_expr(
+            'va.AC = gs.map(g => g.GT.gt).sum(), va.nCalled = gs.filter(g => isDefined(g.GT)).count()')
+               .filter_variants_expr('va.AC > 0 && va.AC < 2 * va.nCalled').persist())
+        n_variants = vds.count_variants()
+        if n_variants == 0:
+            raise FatalError("Cannot run GRM: found 0 variants after filtering out variants that are all homozygous reference or all homozygous variant.")
+        normalized_genotype_expr = '''let mean = va.AC / va.nCalled in
+            if (isDefined(g.GT)) (g.GT.gt - mean) / sqrt(mean * (2 - mean) * {} / 2)
+            else 0'''.format(n_variants)
+
+        f = new_temp_file(suffix="bm")
+
+        vds._jvds.writeBlockMatrix(f, normalized_genotype_expr, BlockMatrix.default_block_size())
+        vds.unpersist()
+
+        bm = BlockMatrix.read(f).cache()
```

Comment:
I don't think the cache should be here anymore with the new block matrix infrastructure.  @jbloom22 @tpoterba?

Reply:
don't know enough about the new infrastructure to answer